### PR TITLE
Add HTML title attribute for each radio input

### PR DESCRIPTION
**Pull request by mekhtiari**
It's useful  in some cases like 5star rating widgets.

### CODE REVIEW

File: lib/Cake/View/Helper/FormHelper.php [hunk 1]
```diff
@@ -1484,6 +1484,7 @@ public function checkbox($fieldName, $options = array()) {
  * - `disabled` - Set to `true` or `disabled` to disable all the radio buttons.
  * - `empty` - Set to `true` to create an input with the value '' as the first option. When `true`
  *   the radio label will be 'empty'. Set this option to a string to control the label value.
+ *   `titles` - define an array of 'radio value' => 'HTML title attribute' to set each radio HTML title attribute
```
Comment (jadb) [inline lib/Cake/View/Helper/FormHelper.php]: Missing a leading `-` for the bullet point list.

**Comment by markstory** [comment]
Instead of adding a new mode, we should probably re-use the expanded options features that `select()` provides.

**Comment by mohsen-ekhtiari** [comment]
You mean allowing options to be defined as either [value => text] or something like [value=>['text' => text, 'title' => title]]?
I thought of it at first but then I looked at the code and saw it would be more meaningful to extend attributes parameter as the following lines suggest:
    `foreach ($options as $optValue => $optTitle) { ...`
But if you think so, I could change the name of $optTitle to $optItems and check for whether its array or not.

**Comment by markstory** [comment]
@mohsen-ekhtiari Exactly that. Being able to use 'complex' options will let people define any attributes they want, not just title.

**Comment by dereuromark** [comment]
Closing as PR for the 'complex' options is now open.

**Pull request closed without merging**